scheduler/crates/api/src/calendar/create_calendar.rs

```diff
@@ -92,7 +94,14 @@ impl UseCase for CreateCalendarUseCase {
     const NAME: &'static str = "CreateCalendar";
 
     async fn execute(&mut self, ctx: &NettuContext) -> Result<Self::Response, Self::Error> {
-        let user = match ctx.repos.users.find(&self.user_id).await {
+        let user = ctx
+            .repos
+            .users
+            .find(&self.user_id)
+            .await
+            .map_err(|_| UseCaseError::InternalError)?;
```

Comment:
Note this `?` at the end

It allows to return directly the error. This is equivalent to the following

```rs
let user = match ctx
            .repos
            .users
            .find(&self.user_id)
            .await
            .map_err(|_| UseCaseError::InternalError) {
    Ok(user) => user,
    Err(e) => return Err(e), // Early return in the function     
 };
```